mathgenerator/mathgen.py

```diff
@@ -36,9 +36,17 @@ def subtractionFunc(maxMinuend = 99, maxDiff = 99):
     return problem, solution
 
 def multiplicationFunc(maxRes = 99, maxMulti = 99):
-    a = random.randint(0, maxMulti)
-    b = random.randint(0, min(maxRes, maxMulti))
-    c = a*b
+    c = random.randint(0, maxMulti) # Generate the solution to the problem
+    i = int(1)
+    j = 0 #Indices for the loop
+    d = [] #Array to store the factors
+    while(i<=c): #Factors are less than or equal to the number
+        if(c%i == 0): #If the remainder is 0, the number is a factor
+            d[j] = i #Store the factor
```

Comment:
IndexError: list assignment index out of range. I'd suggest d.append(i) and forget about j as it simply increases by 1 each time. However, when I tried implementing this, I got stuck in an endless loop because of another error somewhere in your while loop. I'd suggest putting `print(multiplication())` at the bottom of the file and testing it. 
An easier and more efficient way to solve this issue would be to just do something like this:
```
a = random.randint(0, maxMulti)
b = random.randint(0, min(int(maxRes/a), maxMulti))
c = a*b
```

Reply:
I'm going to close this pr but feel free to open another if you get this fixed.

Reply:
I'm going to go with what you're suggesting.

Reply:
It was just an indentation fault. But I'll go with your suggestion as it is shorter.

Reply:
I have implemented what you're suggesting. At first I thought we would be looking to obtain a particular result and the error was just an extra tab on the i increment line. Thank you